src/naive_bayes.cpp

```diff
@@ -0,0 +1,18 @@
+// -*- mode: C++; c-indent-level: 4; c-basic-offset: 4; indent-tabs-mode: nil; -*-
+
+#include <RcppMLPACK.h>				// MLPACK, Rcpp and RcppArmadillo
+
+#include <mlpack/methods/naive_bayes/naive_bayes_classifier.hpp> 	// particular algorithm used here
+
+// [[Rcpp::export]]
+Rcpp::List naive_bayes(const arma::mat& train, const arma::mat& test, const arma::Row<size_t>& labels, const int& classes) {
+    
+    arma::Row<size_t> results;
+
+    // Initialize with the default arguments.
+    mlpack::naive_bayes::NaiveBayesClassifier<> nbc(train, labels, classes);
+    
+    nbc.Classify(test, results);
+
+    return Rcpp::List::create(Rcpp::Named("results") = results);
```

Comment:
I suggested to remove the object files. 

RStudio has a 'More -> Clean & Rebuild'  entry.

If your tools get in your way, change your tools.